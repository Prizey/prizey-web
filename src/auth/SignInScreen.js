import React, { useEffect } from 'react'
import { Redirect } from '@reach/router'
import { SignIn } from 'croods-auth'
import { SignIn as SignInForm } from 'seasoned-auth-forms-web'

import Layout from 'components/Layout/Layout'

export const Form = ({ create, creating, error }) => (
  <Layout>
    <SignInForm onSubmit={create} submitting={creating} submitError={error} />
  </Layout>
)

export default ({ currentUser, navigate, ...props }) => {
  useEffect(() => {
    if (currentUser) {
      const queryString = new URLSearchParams(window.location.search)
      navigate(queryString.get('next') || '/', {
        replace: true,
      })
    }
  }, [currentUser, navigate])

  return (
    <SignIn
      currentUser={currentUser}
      navigate={navigate}
      {...props}
      render={renderProps => <Form {...renderProps} />}
      renderCreated={() => <Redirect to="/" noThrow />}
    />
  )
}
