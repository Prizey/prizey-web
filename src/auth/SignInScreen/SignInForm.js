import React from 'react'
import { withStyles } from '@material-ui/core/styles'
import { SignIn } from 'seasoned-auth-forms-web'

import SubmitButton from 'design/SubmitButton/SubmitButton'
import SecondaryLink from 'design/SecondaryLink'

const styles = theme => ({
  links: {
    display: 'flex',
    justifyContent: 'space-between',
    marginTop: theme.spacing.sm,
  },
})

export const FormBottom = withStyles(styles)(({ classes, ...props }) => (
  <React.Fragment>
    <SubmitButton {...props} label="LOGIN" />
    <div className={classes.links}>
      <SecondaryLink to="/sign-up" label="Register" />
      <SecondaryLink to="/forgot-password" label="Forgot your password?" />
    </div>
  </React.Fragment>
))

export default props => (
  <SignIn
    {...props}
    renderButton={buttonProps => <FormBottom {...buttonProps} />}
  />
)
