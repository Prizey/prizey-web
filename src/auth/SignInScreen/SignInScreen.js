import React, { useEffect } from 'react'
import { Redirect } from '@reach/router'
import { SignIn } from 'croods-auth'
import SignInForm from './SignInForm'
import createAuthForm from '../AuthForm'

const AuthForm = createAuthForm(SignInForm, 'Login')

export default ({ currentUser, navigate, ...props }) => {
  useEffect(() => {
    if (currentUser) {
      const queryString = new URLSearchParams(window.location.search)
      navigate(queryString.get('next') || '/', {
        replace: true,
      })
    }
  }, [currentUser, navigate])

  return (
    <SignIn
      currentUser={currentUser}
      navigate={navigate}
      {...props}
      render={renderProps => <AuthForm {...renderProps} />}
      renderCreated={() => <Redirect to="/" noThrow />}
    />
  )
}
