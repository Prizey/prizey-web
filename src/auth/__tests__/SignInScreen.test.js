import React from 'react'
import renderer from 'react-test-renderer'

import SignInScreen from '../SignInScreen'

jest.useFakeTimers()
jest.mock('croods-auth', () => ({
  SignIn: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ user: { id: 1 } })}</div>
    </div>
  ),
}))

jest.mock('seasoned-auth-forms-web', () => ({
  SignIn: props => <div {...props}>SignIn</div>,
}))

jest.mock('@reach/router', () => ({
  Link: props => <div {...props}>Link - {props.children}</div>,
  Redirect: props => <div {...props}>Redirect - {props.children}</div>,
}))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - <div>render - {props.render(props)}</div>
      <div>renderCreated - {props.renderCreated({ id: 1 })}</div>
    </div>
  ),
}))

it('renders correctly', () => {
  const tree = renderer.create(<SignInScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirect correctly', () => {
  const params = {
    currentUser: true,
    navigate: jest.fn(),
  }
  renderer.create(<SignInScreen {...params} />).getInstance()

  jest.runAllTimers()
  expect(params.navigate).toHaveBeenCalledTimes(1)
})
