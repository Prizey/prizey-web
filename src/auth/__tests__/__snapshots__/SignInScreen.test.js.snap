// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  render={[Function]}
  renderCreated={[Function]}
>
  New - 
  <div>
    render - 
    <div
      className="Component-root-5"
    >
      <div
        className="Component-header-4"
      />
      <div
        className="Component-container-3"
      >
        <h2
          className="MuiTypography-root-6 MuiTypography-h2-19 MuiTypography-alignCenter-29"
        >
          PRIZEY
        </h2>
        <div
          className="Component-caption-1"
        />
        <div
          className="Component-children-2"
        >
          <div>
            SignIn
          </div>
        </div>
        <p
          className="MuiTypography-root-6 MuiTypography-body1-15 MuiTypography-alignCenter-29"
        />
      </div>
    </div>
  </div>
  <div>
    renderCreated - 
    <div
      noThrow={true}
      to="/"
    >
      Redirect - 
    </div>
  </div>
</div>
`;
